runtime/signals.c

```diff
@@ -465,54 +468,67 @@ CAMLexport int caml_rev_convert_signal_number(int signo)
   return signo;
 }
 
-void * caml_init_signal_stack(void)
+stack_t * caml_init_signal_stack(void)
 {
 #ifdef POSIX_SIGNALS
-  stack_t stk;
-  stk.ss_flags = 0;
-  stk.ss_size = SIGSTKSZ;
+  stack_t * stk = malloc(sizeof(stack_t));
```

Comment:
We have to check for a `malloc` failure here, but it would be preferable not to allocate in a signal handler at all when we are not using mmap for stacks.

I think we need to preserve the non-USE_MMAP_MAP_STACK codepath to keep a stack allocated `stk`, and only modify the behaviour for the MMAP case.

Reply:
hmm. I don't like the malloc either. I would have preferred to use types like in the interface to allow stack-allocation:
``` C
CAMLextern int caml_init_signal_stack(stack_t *);
CAMLextern void caml_free_signal_stack(stack_t *);
```
I hope this wouldn't break user code, because those functions are newly exported.

Reply:
I agree. I would prefer to use memory from stack, too.